tests/database/test_journal_db.py

```diff
@@ -194,3 +196,23 @@ def test_returns_key_from_underlying_db_if_missing(journal_db, memory_db):
     assert memory_db.exists(b'1')
 
     assert journal_db.get(b'1') == b'test-a'
+
+
+def test_journal_db_diff_application_mimics_persist(journal_db, memory_db):
```

Comment:
This seems like something that is ideal for a property based test which does more in depth excercising of the various orders of key deletions, additions and updates.